src/gd_color.c

```diff
@@ -4,21 +4,26 @@
 
 #include "gd.h"
 #include "gd_color.h"
+#include <math.h>
 
 /**
+ * Checks whether two given colors match according to a given threshold in
+ * range [0..100]. dist is the euclidian distance of the color channels in
+ * range [0..2].
+ *
  * The threshold method works relatively well but it can be improved.
  * Maybe L*a*b* and Delta-E will give better results (and a better
  * granularity).
  */
 int gdColorMatch(gdImagePtr im, int col1, int col2, float threshold)
 {
-	const int dr = gdImageRed(im, col1) - gdImageRed(im, col2);
-	const int dg = gdImageGreen(im, col1) - gdImageGreen(im, col2);
-	const int db = gdImageBlue(im, col1) - gdImageBlue(im, col2);
-	const int da = gdImageAlpha(im, col1) - gdImageAlpha(im, col2);
-	const int dist = dr * dr + dg * dg + db * db + da * da;
+    const float dr = (gdImageRed(im, col1) - gdImageRed(im, col2)) / (float) gdRedMax;
+    const float dg = (gdImageGreen(im, col1) - gdImageGreen(im, col2)) / (float) gdGreenMax;
+    const float db = (gdImageBlue(im, col1) - gdImageBlue(im, col2)) / (float) gdBlueMax;
+    const float da = (gdImageAlpha(im, col1) - gdImageAlpha(im, col2)) / (float) gdAlphaMax;
```

Comment:
I've just learned that [float division might be particularly slow](http://ithare.com/infographics-operation-costs-in-cpu-clock-cycles/). Should check whether this is an issue (`gdColorMatch()` is called in a tight loop).